### PR TITLE
Add affinity-based type coercion for seek and comparison operation

### PR DESCRIPTION
**Pull request by krishvishal**
This PR implements SQLite-compatible affinity-based type coercion for seek operations and comparison operators, bringing Limbo's behavior closer to SQLite's type conversion rules.

### Key Changes
- Added affinity handling to all comparison operators (`Eq`, `Ne`, `Lt`, `Le`, `Gt`, `Ge`).
- Added `get_expr_affinity()` to determine expression affinity based on column types and casts.
- Added affinity functionality `CmpInsFlags` which carries the affinity information from translation layer to execution layer.
- Added comprehensive `apply_numeric_affinity()` function that handles many edge cases when converting to numerical quantities.
- Implemented `comparison_affinity()` that follows SQLite's affinity precedence rules.
- Added fuzz tests for partial numeric strings, hex literals, and various numeric formats.

### SQLite Compatibility Improvements
This implementation now correctly handles cases like:

- `SELECT * FROM t WHERE x < X'41'` (blob comparisons)
- `SELECT * FROM t WHERE x < '123abc'` (partial numeric strings)
- `SELECT * FROM t WHERE x < 4.9 ` (float precision in seeks)
- Mixed-type comparisons with proper affinity conversion

Thanks 🤝@PThorpe92 and @pedrocarlo for helping improve my planner understanding.

Closes: https://github.com/tursodatabase/limbo/issues/1607

### CODE REVIEW

**Comment by krishvishal** [comment]
The failed action is because of timeout after 10 minutes.

File: tests/integration/fuzz/mod.rs [hunk 1]
```diff
@@ -69,28 +69,101 @@ mod tests {
...
+        values.push(i64::MAX.to_string());
+        values.push(i64::MIN.to_string());
+        values.push("0".to_string());
+
+        for _ in 0..5 {
+            let val: f64 = rng.random_range(-10000.0..10000.0);
+            values.push(val.to_string());
+        }
+
+        values.push("NULL".to_string()); // Man's greatest mistake
```
Comment (el-yawd) [inline tests/integration/fuzz/mod.rs:121]: LOL

File: core/translate/expr.rs [hunk 2]
```diff
@@ -2198,7 +2216,24 @@ fn emit_binary_insn(
...
+        op,
+        ast::Operator::Equals
+            | ast::Operator::NotEquals
+            | ast::Operator::Less
+            | ast::Operator::LessEquals
+            | ast::Operator::Greater
+            | ast::Operator::GreaterEquals
+            | ast::Operator::Is
+            | ast::Operator::IsNot
+    ) {
```
Comment (PThorpe92) [inline core/translate/expr.rs]: nit: this feels like it would make a good method on ast::Operator :)

Comment (krishvishal) [reply core/translate/expr.rs]: Done

File: tests/integration/fuzz/mod.rs [hunk 3]
```diff
@@ -69,28 +69,101 @@ mod tests {
...
+                    }
                 }
             }
         }
     }
 
+    fn generate_random_comparison_values(rng: &mut ChaCha8Rng) -> Vec<String> {
+        let mut values = Vec::new();
+
+        for _ in 0..5 {
```
Comment (jussisaurio) [inline tests/integration/fuzz/mod.rs]: There should be way more of these than 5. This test is invaluable for catching bugs where some edge case with btree interior nodes causes a bug. It's probably overkill to do the whole `for max in 0..=2000` thing as this test currently does, but still.

Comment (jussisaurio) [reply tests/integration/fuzz/mod.rs]: Apart from the above (plus a recent merge conflict) this looks good to go @krishvishal

Comment (krishvishal) [reply tests/integration/fuzz/mod.rs]: Done.

Increased this value to 1000 and reduced the rows to 100.

**Comment by krishvishal** [comment]
@jussisaurio I've resolved the merge conflicts. Tests are passing. This is ready to be merged.
